### PR TITLE
Add article about continuous integration and deployment on ios

### PR DESCRIPTION
**Pull request by nickskull**

### CODE REVIEW

File: source/_posts/2016-08-26-Continuous-Integration-on-iOS-with-HockeyApp-and-Testflight-Deployment.md [hunk 1]
```diff
@@ -1,7 +1,165 @@
...
-date: 2016-08-26 09:51:27
-tags:
+- iOS
 ---
+
+
+# Continuous Integration With Deployment to TestFlight and HockeyApp
+
+
+We've recently revised our development process on the iOS team and set up a continuos integration server to make and deploy our builds faster, easier and more consistently. This post should shed some light on what we chose and how we set it up.
```
Comment (mariusc) [inline source/_posts/2016-08-26-Continuous-Integration-on-iOS-with-HockeyApp-and-Testflight-Deployment.md]: *continuous

File: source/_posts/2016-08-26-Continuous-Integration-on-iOS-with-HockeyApp-and-Testflight-Deployment.md [hunk 2]
```diff
@@ -1,7 +1,165 @@
...
+
+![Jenkins](https://d32wt9uxwfssua.cloudfront.net/tech-blog/uploads/jenkins_logo.png)
+
+Since a few of us had experience with **Jenkins** it was an obvious first choice and we got our prototype version running in a few days. Jenkins is the king of customizability and features and the industry standard for large projects with complex configurations. 
+
+We were quite happy with our setup, only major issues were the initial setup time for each project and overall *scariness* of the user interface. **It was important that everybody could customize their project without feeling like they are launching a nuclear missile.**
+
+![GitLab](https://d32wt9uxwfssua.cloudfront.net/tech-blog/uploads/gitlab.png)
+
+We wanted to choose something that would be easy to understand even for our interns or junior developers and since we recently updated our installation of GitLab to the latest version, we've been intrigued to try **GitLab's Pipelines** which are specifically meant for continuos integration.
```
Comment (mariusc) [inline source/_posts/2016-08-26-Continuous-Integration-on-iOS-with-HockeyApp-and-Testflight-Deployment.md]: *continuous

File: source/_posts/2016-08-26-Continuous-Integration-on-iOS-with-HockeyApp-and-Testflight-Deployment.md [hunk 3]
```diff
@@ -1,7 +1,165 @@
...
+
+Rewriting of some of the scripts from Jenkins was needed, but worth the result. We've achieved an extremely easy project setup (just adding one file to the repo) and it's easy to understand for everyone. For more complex projects we allow overriding the build scripts per project, again by just putting them in the repo.
+
+**TL;DR:** GitLab's Pipelines won for us. Jenkins is scary.
+
+## Our Setup
+
+### Commit Message Trigger
+
+We figured out that the simplest way of triggering a CI build for us will be some magic words in a commit message. We still need to manually incerment build and version numbers of our apps, and that's the perfect commit where to make the CI build the project.
```
Comment (mariusc) [inline source/_posts/2016-08-26-Continuous-Integration-on-iOS-with-HockeyApp-and-Testflight-Deployment.md]: *increment

File: source/_posts/2016-08-26-Continuous-Integration-on-iOS-with-HockeyApp-and-Testflight-Deployment.md [hunk 4]
```diff
@@ -1,7 +1,165 @@
...
+
+First, we tell the CI where it should deploy the app after building and then the environment the should should be set and used when building the app.
```
Comment (mariusc) [inline source/_posts/2016-08-26-Continuous-Integration-on-iOS-with-HockeyApp-and-Testflight-Deployment.md]: remove one of the shoulds in "should should"

**Comment by nickskull** [comment]
@mariusc Thanks! Fixed it.

**Comment by nickskull** [comment]
@dennishn Can I get someone to review and merge it when ready? :)

**Pull request merged**